README.md

```diff
@@ -1,208 +1,262 @@
-<img src="assets/images/gcm-step4.PNG"/>
+### Step 4 - Send a message
 
-### Step 5 - Create server keys
-Click on the menu option credentials, located on the left hand side under the APIs & Auth section. Then click on the button under the public API Access that says create new key. When the popup appears press the Server key button and then press create on the next screen. Take note of the API Key because you will need this when we set up the push notifcation in Mendix.
-<img src="assets/images/gcm-step5.PNG"/>
+Fill in the title and the message on the form and press `Send immediate` button. You should see a new notification coming in your device. By default, if your hybrid mobile app is currently active, the message will be displayed in the app. Otherwise, it will be shown as a standard push notification.
 
-### Step 6 - Setup Mendix app
-Open up your application in Mendix and login as an Admin, so that you can see the menu option google admin.
-Enter the project ID into the sender ID field and the API key into the API field. 
-<img src="assets/images/gcm-step6.PNG"/>
+<img src="assets/images/testing/send message.JPG"/>
 
-Once entered tick the checkbox enabled and press restart. From now on your application will always start the gcm push notification system for you.
+## APIs
 
-For more information on setting up your google API then please refer to this article: (Google API Setup) [http://developer.android.com/google/gcm/gs.html]
+This is a list of microflows that can be called from within your application. They are located in the `_USE ME/Microflows` folder.
 
+|Microflow name         |Description                                                    |
+|-----------------------|---------------------------------------------------------------|
+|MessageFactory         |A microflow to create a `Message` object                       |
+|SendMessageImmediately |A microflow to send a message to GCM/APNs immediately          |
+|SendMessageQueued      |A microflow to send a message to GCM/APNs using a queue        |
+|SendMessagesImmediately|A microflow to send multiple messages to GCM/APNs immediately  |
+|SendMessagesQueued     |A microflow to send multiple messages to GCM/APNs using a queue|
 
-## Setting up Windows
-Windows requires no additional configuration for push notifications to work. Simply load the application onto a Windows 8 Phone and login to the application. The user's Windows Phone credentials should appear within the device menu.
-The notifications for Windows work using a web service, the widget within your application registers the user's unique URL. With the URL the Mendix application can pass a message to this URL.  
+Note: for sending message(s) queue to work properly, you need to ensure that the scheduled event `PushQueue` is active. In a sandbox/free app this does not require any extra step, but in a TAP environments you will need to explicitly set the scheduled event to be activated.
```

Comment:
Suggestion : 'for sending queued messages to work correctly' 

Q: Where to activate this scheduled event?
Also, maybe note that for sand box/free app, it's determined by the setting in the model?

Finally, elsewhere we use Free App I believe, so let's be consistent there.